.github/workflows/image-and-helm-publish-check-deploy-on-push-scheduled.yml

```diff
@@ -142,14 +142,39 @@ jobs:
       image_tag: ${{ github.ref_name == 'main' && '' || needs.create_branch_identifier.outputs.image_tag_from_branch }}
       helm_chart_version_generation: ${{ needs.select_helm_version_generation_and_image_tag_generation.outputs.SELECT_HELM_VERSION_GENERATION }}
       helm_chart_version: ${{ github.ref_name == 'main' && '' || needs.create_branch_identifier.outputs.chart_version_from_branch }}
+    
 
   wait_for_helm_chart_to_get_published:
+    runs-on: ubuntu-latest
     needs:
+      - create_branch_identifier
       - release_helm
-    runs-on: ubuntu-latest
     steps:
-      - shell: bash
-        run: sleep 1m 
+      - name: Check for Helm chart
+        run: |
+          CHART_NAME=schulportal-client
+          CHART_VERSION="${{ needs.release_helm.outputs.helm_chart_version }}"
+          echo "CHART_VERSION=$CHART_VERSION"
+          REPO_URL="https://dbildungsplattform.github.io/helm-charts-registry/"
+
+          echo "Adding Helm repository $REPO_URL"
+          helm repo add dbildungsplattform $REPO_URL
+          
+          echo "Checking for chart $CHART_NAME version $CHART_VERSION in $REPO_URL"
+
+          for i in {1..20}; do
+            echo "Attempt $i..."
+            helm repo update
+            if helm search repo dbildungsplattform/$CHART_NAME --version $CHART_VERSION | grep -qw "$CHART_VERSION"; then
+              echo "Chart $CHART_NAME-$CHART_VERSION found after $i seconds !"
```

Comment:
I think `i` has to be multiplied with 30 seconds if we want to output the time here.